### DISCUSSION TITLE
New website

### DISCUSSION THREAD

**Discussion by EduardoLopes** [rome/tools#1518] [General]
I did most of the work in the current website, and when i did, i was in a moment where i was getting back to programming after almost 2 years without programming. There's thousands of things i can do better today and i would like to start to work on the new website. I'm opening this discussion, so we can define how it will be made!

I have suggested on [discord](https://discord.com/channels/678763474494423051/678763474930761739/826159225360875630) to drop eleventy. And i don't think we need to use no deps and Rome for everything in the website now, but once Rome is ready!

## Framework

In a local branch, I've already started to make some experiments using nextjs. Even though i like nextjs, i'm not sure it's the best option and we should consider the others frameworks. Lately I've been making a tool using [SvelteKit](https://kit.svelte.dev/) and i like it. There's also [gridsome](https://gridsome.org/), but looks like it'll take some time to [upgrade to vue 3](https://github.com/gridsome/gridsome/issues/1289).

Even though i think we should use a framework, we still should keep deps at a minimum.

## Styles

I would like to have styles centralized in one place, so any Rome webpage can use it. At first, service and website/docs! For this to happen, the front-end of the Rome service should use the same thing for styles as the website. I imagine sass or any [css-in-js solution](https://vanilla-extract.style/). 

The module with all the styles would define general styles like color, spacing, typography, animation and etc, following the [design tokens](https://spectrum.adobe.com/page/design-tokens/) philosophy. Any specific style would be defined in the project itself, not in the module with general styles.

I don't like to use css vars for that because i think `color.primary("light");` (scss or js) is better to read than `var(--color-primary-light)`. In some cases the var can get too big!

I've done some experiments with this idea a while back, but it can be done better than [this].(https://github.com/EduardoLopes/design-tokens-study/blob/main/src/styles/_space.scss)!

I can do a proof of concept, if needed!

## Pages

Rome will be lot's of tools and i don't like the idea of having everything documented in a single page. Let's have a page for each tool!

----

There might be some points I'm missing! But I'm bringing this to discussion because i don't like the current website anymore and i wanna start working on a better visual identity to make a good looking/modern website. I'll be happy with anything we decided to use!

**Comment by UsamaHameed** [comment]
Do we already have a design in the works? Or do we need to work on that as well?

**Reply by EduardoLopes** [reply]
We have basically this: https://github.com/rome/tools/tree/main/assets

So we still need to work on that

**Reply by UsamaHameed** [reply]
So basically just colours, icons and typography. We need to work on the prototype.

**Reply by jamiebuilds** [reply]
We've been talking about contracting a designer to help us set our early brand identity. We could also have them provide us with a website design

**Reply by EduardoLopes** [reply]
Ohhh. I don't consider me a designer, i just know some stuff about it! and have a professional doing that would be awesome! ✨

**Comment by EduardoLopes** [comment]
Considerations about the design of the current website:

- Too much borders. There's borders everywhere and we should use space in most places
- It's hard to work with yellow (the primary color) on white
- It looks like it was made years ago

Anything else we could add to this list?

**Reply by UsamaHameed** [reply]
I think we can increase spacing between the elements in the sidebar. Its too clunky and puts me off whenever I wanna read something. 
We can also have better styles for links. For example, use an icon next to the text, instead of underlining the text. 
The visualization for `Rome is a [carousel] [carousel]` can probably be made much better. The light mode background shouldn't be too white but thats a personal preference.

**Reply by EduardoLopes** [reply]
I wanted link to be yellow, but the contrast isn't good on the white background! set links as bold would fix that, but we removed since the emphasis to links were too much when set to bold. I'm not sure about use an icon next to the text, but we can experiment with that.

> The visualization for `Rome is a [carousel] [carousel]` can probably be made much better. The light mode background shouldn't be too white but that's a personal preference.

Yeah. we can improve that

**Reply by UsamaHameed** [reply]
Something like: 
Home (paperclip or chainlink icon here) 

https://fontawesome.com/icons/link?style=solid

and we don't apply much text decoration to the link text. The icon makes it clear that its a link and at the same time, no text decoration is easier on the eyes.

**Reply by EduardoLopes** [reply]
Do you know any website that do that? I already saw something like this somewhere, but i can't remember

**Reply by UsamaHameed** [reply]
Yes, we use it at lisk (I work for them :D) 
- https://lisk.com/docs 
- https://lisk.com/styleguide

**Reply by UsamaHameed** [reply]
![Screenshot 2021-05-19 at 20 32 45](https://user-images.githubusercontent.com/11444352/118865592-65342280-b8e1-11eb-9ff5-24ba442b72a3.png)

**Reply by EduardoLopes** [reply]
Nice. this looks good. But it's only for links with `target="_blank"`?

**Reply by UsamaHameed** [reply]
Hmm yes. It wouldn't make sense to have that icon for simple links. I didn't think about that. I guess we can't add an icon for `target=_self`

**Reply by EduardoLopes** [reply]
I did this to see how it would look and i think it's too noisy.

![Screenshot 2021-05-19 164451](https://user-images.githubusercontent.com/1084297/118874702-9270d880-b8c1-11eb-873a-2a60cdca8881.png)

But this makes a lot of sense for links with `target="_blank"`

**Reply by UsamaHameed** [reply]
I agree. Also, the noise really is the fact that we have too many links in a rather short text :D

**Reply by jamiebuilds** [reply]
@sebmck and I spoke about that `Rome is...` section. We are in favor of getting rid of it from the homepage since it's unlikely to drive adoption once Rome v1 is released

**Comment by EduardoLopes** [comment]
We should have an topbar for important links and search. I don't like the sidebar only design.

**Comment by jer3m01** [comment]
Personally more a fan of Preact, although I'm not against using Vue or Svelte just don't have experience with it. Don't know about how the rest of the team feels.

Not trying to push against anything other than preact/react but using a tool the team is more comfortable with is probably a good idea. More so the staff as they'll probably be maintaining the website for longer.

**Reply by UsamaHameed** [reply]
+1 I don't mind having to use any of the mentioned libs but probably best to use the one we are most comfortable with. Unless we have some dedicated devs who focus on the website only. I guess most of us wanna work with the rest and not the website

**Reply by EduardoLopes** [reply]
Yeah, i mentioned the others tools just to see what people think! And I'm inclined to use React. I've never used Preact, but it's basically React, so I'm ok with that!

And I'll be working mostly focused on the website/front-ends only!

**Comment by jamiebuilds** [comment]
Please drop any links to developer tools / services websites that you really like. Personally I really like https://tailwindcss.com/ I would love to have interactive sections that explain the features of Rome really well

**Reply by EduardoLopes** [reply]
I really like the interactive sections! I was imagine something like that for the website! 

There's also the github homepage: https://github.com/ (logout or use browser's private mode to see)

**Reply by UsamaHameed** [reply]
I really like the idea of having interactivity like tailwind and github but looking at those pages, I think whats more important is that we get a professional design made. I understand this is already in the works so 💯
